api/src/main/java/com/netflix/iceberg/transforms/Transforms.java

```diff
@@ -56,12 +58,27 @@ private Transforms() {
     if (transform.equalsIgnoreCase("identity")) {
       return Identity.get(type);
     } else if (type.typeId() == Type.TypeID.TIMESTAMP) {
-      return Timestamps.valueOf(transform.toUpperCase(Locale.ENGLISH));
+      return match(
+          () -> Timestamps.valueOf(transform.toUpperCase(Locale.ENGLISH)),
+          () -> new UnknownTransform<>(type, transform)
+      );
     } else if (type.typeId() == Type.TypeID.DATE) {
-      return Dates.valueOf(transform.toUpperCase(Locale.ENGLISH));
+      return match(
+          () -> Dates.valueOf(transform.toUpperCase(Locale.ENGLISH)),
+          () -> new UnknownTransform<>(type, transform)
+      );
     }
 
-    throw new IllegalArgumentException("Unknown transform: " + transform);
+    return new UnknownTransform<>(type, transform);
+  }
+
+  private static Transform<?,?> match(Supplier<Transform> transformSupplier,
```

Comment:
Nit: arguments should start at the same column or should be wrapped starting at 4 spaces from `private`.